### ISSUE TITLE
Not work with exotic object

### ISSUE THREAD

**Issue by Jack-Works** [salesforce/near-membrane#384]
Reproduction:

```js
import { AsyncCall } from 'https://cdn.skypack.dev/async-call-rpc@6.1.2'
import createIframeVirtualEnvironment from 'http://cdn.skypack.dev/@locker/near-membrane-dom'

const exoticObject = AsyncCall(null, {
    channel: { on() {}, send() {} },
})

const vm = createIframeVirtualEnvironment(globalThis)

debugger
const obj = vm.evaluate(`e => e.foo`)(exoticObject)

// obj should be a function.
```

**Comment by Jack-Works** [comment]
fixed on async-call-rpc side by implementing a [[GetOwnPropertyDescriptor]] trap, but this problem still happens with other exotic objects.

**Comment by jdalton** [comment]
Hi @Jack-Works!

By exotic objects do you mean proxied objects?

**Comment by Jack-Works** [comment]
Yes. A Proxy that implements Get but not GetOwnPropertyDescriptor

**Comment by jdalton** [comment]
Closed by #392.
